qa/basicTest.py

```diff
@@ -0,0 +1,103 @@
+        #check variable configuration
+        assert_true(sc_nodes > 0, "Sidechain nodes must be > 0")
+        assert_true(len(os.getenv("BITCOINCLI", "")) > 0, "BITCOINCLI env var must be set")
+        assert_true(len(os.getenv("BITCOIND", "")) > 0, "BITCOIND env var must be set")
+        assert_true(len(os.getenv("SIDECHAIN_SDK", "")) > 0, "SIDECHAIN_SDK env var must be set")
+        assert_true(len(os.getenv("APP_JAR", "")) > 0, "APP_JAR env var must be set")
+        assert_true(len(os.getenv("APP_LIB", "")) > 0, "APP_LIB env var must be set")
+        assert_true(len(os.getenv("APP_MAIN", "")) > 0, "APP_MAIN env var must be set")
+        print "Configuration of this test:"
+        print "    1 mainchain node"
+        print "    {} sidechains node".format(sc_nodes)
+        self.number_of_mc_nodes = 1
+        self.number_of_sidechain_nodes = sc_nodes
+
+    def setup_chain(self):
+        initialize_chain_clean(self.options.tmpdir,  self.number_of_mc_nodes)
+
+    def setup_network(self, split = False):
+        print("Initializing Mainchain nodes...")
+        self.nodes = self.setup_nodes()
+        self.sync_all()
+        print("OK\n")
+
+    def setup_nodes(self):
+        return start_nodes(self.number_of_mc_nodes, self.options.tmpdir)
+
+    def sc_setup_chain(self):
+        mc_node_1 = self.nodes[0]
+
+        sc_node_configuration = []
+        for x in range(1,  self.number_of_sidechain_nodes +1):
+            sc_node_configuration.append(
+                SCNodeConfiguration(
+                    MCConnectionInfo(address="ws://{0}:{1}".format(mc_node_1.hostname, websocket_port_by_mc_node_index(0)))
+                )
+            )
+        network = SCNetworkConfiguration(SCCreationInfo(mc_node_1, 600, 1000), *sc_node_configuration)
+        bootstrap_sidechain_nodes(self.options.tmpdir, network)
+
+
+
+    def sc_setup_network(self, split = False):
+        print("Initializing {} Sidechain nodes...".format(self.number_of_sidechain_nodes))
+        self.sc_nodes = self.sc_setup_nodes()
+        if (self.number_of_sidechain_nodes > 1):
+            for x in range(1,  self.number_of_sidechain_nodes):
+                print('Connecting sidechain node node0 with node{}'.format(x))
+                connect_sc_nodes(self.sc_nodes[0], x)
+        self.sc_sync_all()
+        print("OK\n")
+
+    def sc_setup_nodes(self):
+        lib_separator = ":"
+        if sys.platform.startswith('win'):
+            lib_separator = ";"
+        app_jar = os.getenv("APP_JAR", "")
+        app_lib = os.getenv("APP_LIB", "")
+        app_main = os.getenv("APP_MAIN", "")
+
+        path = []
+        for x in range(1,  self.number_of_sidechain_nodes + 1):
+            path .append(app_jar + lib_separator + app_lib + " " + app_main)
+
+        return start_sc_nodes(self.number_of_sidechain_nodes, self.options.tmpdir, None, None, path)
+
+    #spends initial forgin stake to create a stadard coinbox
+    def convertInitialForging(self):
```

Comment:
Instead of spending initial forging stake I would suggest to make a forward transfer.
If you will spend the stake and will not create another one, than in 2 epochs there will be no stakes registered for forging, so will not be able to produce new sc blocks.

Reply:
fixed!